std/traits.d

```diff
@@ -8087,3 +8089,105 @@ enum isCopyable(S) = is(typeof(
     static assert(isCopyable!int);
     static assert(isCopyable!(int[]));
 }
+
+private template TypeOf(T...)
+    if (T.length == 1 && (is(T[0]) || is(typeof(T[0]))))
+{
+    static if (is(T[0]))
+        alias TypeOf = T[0];
+    else static if (is(typeof(T[0])))
+        alias TypeOf = typeof(T[0]);
+}
+
+@safe unittest
+{
+    assert(is(TypeOf!3 == int));
+    assert(is(TypeOf!int == int));
+    assert(is(TypeOf!([]) == void[]));
+    assert(!is(TypeOf!(mixin(__MODULE__))));
+    assert(is(TypeOf!TypeOf));
+}
+
+/**
+ * Get a list of the overloads for the given callable.
+ *
+ * Params:
+ *     func = The callable whose overload set to return.
+ * Returns:
+ *     A tuple of the overloads for the given callable.
+ */
+template getOverloads(func...)
+    if (func.length == 1 && isCallable!func)
+{
+    enum isModuleOrPackage(alias s) = !__traits(compiles, { alias _ = typeof(s); });
+    alias Parent = AliasSeq!(__traits(parent, func))[0];
+
+    static if (is(Parent) || isModuleOrPackage!Parent)
+        alias getOverloads = AliasSeq!(__traits(getOverloads, Parent, __traits(identifier, func)));
+    else
+        alias getOverloads = func;
+}
+
+version (unittest)
+{
+    // Local functions can't have overloads, so the functions used by
+    // getOverloads and overloadFor need to be defined here.
+    private:
+    static int getOverloadsTest(int n = 3) { return n; }
```

Comment:
Since you can't add these in the unit test, just add them as three commented lines so the doc reader knows what's being tested.

Reply:
Sorry, I don't quite get what you're asking me to do. You mean each line has to have a separate comment? Doc comment (`///`)?
```
/// For getOverloads and overloadFor
private static int getOverloadsTest(int n = 3) { return n; }
/// Ditto
static int getOverloadsTest(int, int)  { return 3; }
/// Ditto
static int getOverloadsTest2(int, int) { return 3; }
```

Reply:
like this

```
@safe unittest
 {
     // int getOverloadsTest(int n = 3) { return n; }
     // int getOverloadsTest(int, int)  { return 3; }
     // int getOverloadsTest2(int, int) { return 3; }

     assert(getOverloads!getOverloadsTest.length == 2);
     assert(is(typeof(&getOverloadsTest) == typeof(&getOverloads!getOverloadsTest[0])));
     assert(!is(typeof(&getOverloadsTest) == typeof(&getOverloads!getOverloadsTest[1])));
     alias fn = getOverloadsTest;
     assert(is(typeof(&fn) == typeof(&getOverloads!getOverloadsTest[0])));
     assert(!is(typeof(&fn) == typeof(&getOverloads!getOverloadsTest[1])));
     assert(getOverloads!getOverloadsTest2.length == 1);
     assert(getOverloads!(() => 3).length == 1);
     assert(getOverloads!(() => 3).length == 1);
 }
```